### PR TITLE
am62x: secure boot: document the TI key writer

### PR DESCRIPTION
**Pull request by ldts**
## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 3da7e2074261a935d1f681c45f8bc445464e81db are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2206/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for dea6cc4a343e72818a0e5a49b300988dd07e3287 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2207/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for a2213909846444feb6bf264383c9f2709e30ac17 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2208/docs/artifacts/html/index.html

**Comment by quaresmajose** [comment]
https://ci.foundries.io/projects/fio-docs/builds/2213/docs/artifacts/html/reference-manual/boards/am62xx-sk.html
https://ci.foundries.io/projects/fio-docs/builds/2213/docs/artifacts/html/reference-manual/boards/am64xx-sk.html

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+To protect the AES encryption key and the hash two additional extensions
+are created:
+
+   1. AES-256 key
+   2. AES-256 key signature generated using the SMPK private key
+
+These are protected/encrypted with the TI FEK public key (also
+RSA-4096).
+
+All these four extensions are then combined into an X.509 configuration
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:104]:
```suggestion
All four extensions are then combined into an X.509 configuration
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+certificate. This configuration certificate is finally signed with the
+SMPK private key to create the final X509 certificate.
+
+The OTP Keywriter project logically needs access to the keys to be fused
+as well as the TI FEK key
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:109]:
```suggestion
The OTP Keywriter project needs access to the keys to be fused,
as well as the TI FEK key.
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+                | |      U-Boot       | |
+                | +-------------------+ |
+                | |                   | |
+                | |     U-Boot dtb    | |
+                | +-------------------+ |
+                +-----------------------+
+
+
+Signing the different components of the boot-chain was recently
+`integrated`_ in U-Boot's binman therefore simplifying the previous
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:195]:
```suggestion
`integrated`_ in U-Boot's binman, simplifying the previous
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+                | |                   | |
+                | |      U-Boot       | |
+                | +-------------------+ |
+                | |                   | |
+                | |     U-Boot dtb    | |
+                | +-------------------+ |
+                +-----------------------+
+
+
+Signing the different components of the boot-chain was recently
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:194]:
```suggestion
Signing the different components of the boot-chain has been
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+`integrated`_ in U-Boot's binman therefore simplifying the previous
+process; at the time of this writing the code is only available in the
+vendor's repository hence why this page uses hyperlinks to vendor
+software and not upstream.
+
+Compiling U-Boot will take care of signing not only the binaries it
+generates but also the rest of the firmware images that need to be
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:201]:
```suggestion
generates, but the rest of the firmware images that need to be
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+included in the final images
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:202]:
```suggestion
included in the final images.
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+As a user, you'll merely need to **replace** U-Boot's ``customMpk.pem`` with
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:204]:
```suggestion
As a user, you will need to **replace** U-Boot's ``customMpk.pem`` with
```

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+the RSA-4096 key that was fused during provisioning: this will sign all
+other binaries and firmwares including the externally generated
```
Comment (kprosise) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:206]:
```suggestion
other binaries and firmwares, including the externally generated
```

**Review by kprosise — COMMENTED** [review]
Suggestions provided

**Comment by ldts** [comment]
> Suggestions provided

thanks katrina!. all applied

**Comment by doanac** [comment]
Docs for 854ca9d4182fbee6647722f51c3a48edf6fd7b05 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2215/docs/artifacts/html/index.html

File: source/reference-manual/security/secure-boot-ti-am62x.rst [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+`integrated`_ in U-Boot's binman, simplifying the previous
+process; at the time of this writing the code is only available in the
+vendor's repository hence why this page uses hyperlinks to vendor
+software and not upstream.
+
+Compiling U-Boot will take care of signing not only the binaries it
+generates, but the rest of the firmware images that need to be
+included in the final images.
+
+As a user, you will need to **replace** U-Boot's ``customMpk.pem`` with
```
Comment (ricardosalveti) [inline source/reference-manual/security/secure-boot-ti-am62x.rst:204]: custMpk.pem

**Comment by doanac** [comment]
Docs for 07c94a429692dbe97e401ef0e500ef3742fb5571 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2219/docs/artifacts/html/index.html

**Pull request merged**